### DISCUSSION TITLE
Core pagination support for double layered pagination (👍74)

### DISCUSSION THREAD

**Discussion by zachleat** [11ty/eleventy#3987] [Enhancement Queue]
Related issue #332, something with a little bit more DX might be warranted since this has come up a few times now.

**Comment by zachleat** [comment]
This repository is now using [lodash style issue management](https://twitter.com/samselikoff/status/991395669016436736) for enhancements. This means enhancement issues will now be closed instead of leaving them open.

View [the enhancement backlog here](https://github.com/11ty/eleventy/issues?utf8=%E2%9C%93&q=label%3Aneeds-votes+sort%3Areactions-%2B1-desc+). Don’t forget to upvote the top comment with 👍!

**Comment by distinctperspective** [comment]
I converted too. @zachleat  I can help you with that.

**Comment by MWDelaney** [comment]
I needed this kind of thing for a recent project and this is what I came up with. Related to #332, also.

https://gist.github.com/MWDelaney/6dd60be5e4c223217d597983d20bde20

**Comment by jeromecoupe** [comment]
Yup, using the same kind of approach, albeit using custom taxonomies instead of tags:

https://github.com/jeromecoupe/11ty-paginate-taxonomies

**Comment by MWDelaney** [comment]
Expanding the above into a working example...

After much heartache, and many compromises, I have paginated dynamic tag pages for blogging working.

I made a pull request against ZeroPoint demonstrating it:
https://github.com/MWDelaney/ZeroPoint/pull/31

I'm sure this could be simplified, and pagination more directly modified, but it works!

**Comment by dave-kennedy** [comment]
I'm confused by the status of this issue. It's pinned to the top of the issues list, so I would expect a canonical answer here. @zachleat can you please clarify? Is the recommended solution [this one](https://github.com/11ty/eleventy/issues/332#issuecomment-445236776)? If so, it might make more sense to pin that issue instead.

**Comment by ryangittings** [comment]
+1 on this - keeps coming up and although there are snippets we can use, feels like this should be handled better in core!

**Comment by zachleat** [comment]
Cross linking to this comment here, which has an example (still too complicated) of this implemented using the `before` callback option https://github.com/11ty/eleventy/issues/332#issuecomment-1867741229

**Comment by zachleat** [comment]
Cross linking to Nested Collections enhancement queue entry https://github.com/11ty/eleventy/discussions/3991
